filebeat/autodiscover/builder/hints/logs.go

```diff
@@ -37,13 +37,6 @@ import (
 	"github.com/elastic/elastic-agent-libs/mapstr"
 )
 
-func init() {
```

Comment:
So we can't use `init()` anymore without side effects for agentbeat? 

I don't mind it, but this seems like something that will be forgotten or undone without some kind of automated enforcement.

Reply:
Correct, caution will need to be taken in the case that a beat is modifying a behavior that is coming from libbeat. 

If the change can bleed into another beat then you need to use InitializeModule, otherwise init can be used. 

Like init is still being used in most places, but only the areas where it can cause an effect was it changed to InitializeModule.

Reply:
It is also really hard to enforce with automation, because its conditional if it is required. It all depends on the context of the change.

It applied to anything that is modifying a behavior that is coming from libbeat or x-pack/libbeat.